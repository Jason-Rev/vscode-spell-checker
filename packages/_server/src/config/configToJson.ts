import { CSpellSettings } from '@cspell/cspell-types';

export function configToJson(config: CSpellSettings): string {
    function replacer(_key: string, value: unknown) {
        if (value instanceof RegExp) {
            return value.toString();
        }
        return value;
    }
    return JSON.stringify(config, replacer, 2);
}
