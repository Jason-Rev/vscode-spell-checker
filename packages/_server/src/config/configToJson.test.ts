import { CSpellSettings } from 'cspell-lib';
import { configToJson } from './configToJson';

describe('Validate configToJson', () => {
    const sampleConfig: CSpellSettings = {
        version: '0.2',
        patterns: [
            {
                name: 'word',
                pattern: /^\s*word/gim,
            },
        ],
        ignoreRegExpList: [/'.*'/g],
    };

    test.each`
        cfg                 | message
        ${{}}               | ${'empty config'}
        ${{ sampleConfig }} | ${'sampleConfig'}
    `('configToJson $message', ({ cfg }) => {
        expect(configToJson(cfg)).toMatchSnapshot();
    });
});
