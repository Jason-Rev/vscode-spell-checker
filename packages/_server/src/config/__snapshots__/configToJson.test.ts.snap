// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate configToJson configToJson empty config 1`] = `"{}"`;

exports[`Validate configToJson configToJson sampleConfig 1`] = `
"{
  \\"sampleConfig\\": {
    \\"version\\": \\"0.2\\",
    \\"patterns\\": [
      {
        \\"name\\": \\"word\\",
        \\"pattern\\": \\"/^\\\\\\\\s*word/gim\\"
      }
    ],
    \\"ignoreRegExpList\\": [
      \\"/'.*'/g\\"
    ]
  }
}"
`;
